site/pages/view-all-disruptions.page.tsx

```diff
@@ -362,6 +379,30 @@ const ViewAllDisruptions = ({ disruptions, services, newDisruptionId }: ViewAllD
         ); // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [selectedServices]);
 
+    useEffect(() => {
+        const operatorsToSet = selectedOperatorsNocs.map(
```

Comment:
We could avoid iterating twice by adding objects to `disruptionOperatorsToSet` while mapping over `selectedOperatorsNocs`

```const disruptionOperatorsToSet: DisruptionOperator[] = [];
        const operatorsToSet = selectedOperatorsNocs.map((selOpNoc) => {
            const operator = operators.find((op) => op.nocCode === selOpNoc) as Operator;
            if (operator) {
                disruptionOperatorsToSet.push({
                    operatorName: operator.operatorPublicName,
                    operatorRef: operator.nocCode,
                });
                return operator;
            }
        });```

Reply:
Good idea - the code needed tweaking a little because in a map you need to always return a value, and the if statement was avoiding that. I changed it to a forEach to get around it.